<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1104</width>
    <height>497</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Find-Object</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
      <property name="spacing">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,0,0,0,0">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Camera</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_timeRefreshRate">
          <property name="text">
           <string> (0 Hz - 0 Hz)</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_detectorDescriptorType">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_nfeatures">
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string> features</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="ObjWidget" name="imageView_source" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1104</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_objects"/>
    <addaction name="actionSave_objects"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionAdd_object_from_scene"/>
    <addaction name="actionAdd_objects_from_files"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_scene_from_file"/>
    <addaction name="separator"/>
    <addaction name="actionStart_camera"/>
    <addaction name="actionPause_camera"/>
    <addaction name="actionStop_camera"/>
    <addaction name="separator"/>
    <addaction name="actionCamera_from_video_file"/>
    <addaction name="separator"/>
    <addaction name="actionRestore_all_default_settings"/>
    <addaction name="separator"/>
    <addaction name="actionRemove_all_objects"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>?</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_parameters">
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>194</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Parameters</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="ParametersToolBox" name="toolBox">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="page_1">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>276</width>
          <height>357</height>
         </rect>
        </property>
        <attribute name="label">
         <string>Statistics</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0">
           <property name="horizontalSpacing">
            <number>6</number>
           </property>
           <property name="verticalSpacing">
            <number>0</number>
           </property>
           <item row="0" column="2">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>ms</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="label_timeExtraction">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>ms</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Descriptors extraction</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Features detection</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_timeDetection">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Descriptors matching</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Descriptors indexing</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_timeIndexing">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="label_timeMatching">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>ms</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>ms</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Detect outliers and GUI</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="label_timeGui">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>ms</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="label_minMatchedDistance">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Min matched distance</string>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Max matched distance</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="label_maxMatchedDistance">
             <property name="text">
              <string>000</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>174</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_restoreDefaults">
       <property name="text">
        <string>Restore defaults</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_objects">
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>196</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Objects</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QScrollArea" name="objects_area">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>0</height>
        </size>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>274</width>
          <height>389</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_objects">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>230</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_updateObjects">
       <property name="text">
        <string>Update objects</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAdd_object_from_scene">
   <property name="text">
    <string>Add object from scene...</string>
   </property>
  </action>
  <action name="actionStart_camera">
   <property name="text">
    <string>Start camera</string>
   </property>
  </action>
  <action name="actionStop_camera">
   <property name="text">
    <string>Stop camera</string>
   </property>
  </action>
  <action name="actionSave_objects">
   <property name="text">
    <string>Save objects...</string>
   </property>
  </action>
  <action name="actionLoad_objects">
   <property name="text">
    <string>Load objects...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionRestore_all_default_settings">
   <property name="text">
    <string>Restore all default settings</string>
   </property>
  </action>
  <action name="actionAdd_objects_from_files">
   <property name="text">
    <string>Add objects from files...</string>
   </property>
  </action>
  <action name="actionLoad_scene_from_file">
   <property name="text">
    <string>Load scene from file...</string>
   </property>
  </action>
  <action name="actionPause_camera">
   <property name="text">
    <string>Pause camera</string>
   </property>
  </action>
  <action name="actionCamera_from_video_file">
   <property name="text">
    <string>Camera from video file...</string>
   </property>
  </action>
  <action name="actionRemove_all_objects">
   <property name="text">
    <string>Remove all objects</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ObjWidget</class>
   <extends>QWidget</extends>
   <header>ObjWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ParametersToolBox</class>
   <extends>QToolBox</extends>
   <header>ParametersToolBox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
